src/main/java/com/knightcharacter/app/service/impl/SkillServiceImpl.java

```diff
@@ -0,0 +1,64 @@
+package com.knightcharacter.app.service.impl;
+
+import com.knightcharacter.app.domain.BonusVO;
+import com.knightcharacter.app.domain.SkillVO;
+import com.knightcharacter.app.exception.SkillNotFoundException;
+import com.knightcharacter.app.gateway.SkillGateway;
+import com.knightcharacter.app.service.BonusService;
+import com.knightcharacter.app.service.SkillService;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import lombok.RequiredArgsConstructor;
+
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class SkillServiceImpl implements SkillService {
+
+    private final SkillGateway skillGateway;
+    private final BonusService bonusService;
+
+    @Override
+    public SkillVO save(SkillVO skillVO) {
+        skillVO.setBonuses(fetchBonusesByIds(extractBonusIds(skillVO)));
+        return skillGateway.save(skillVO);
+    }
+
+    private List<BonusVO> fetchBonusesByIds(List<String> bonusIds) {
+        return bonusIds.stream().map(bonusService::findById).collect(Collectors.toList());
+    }
+
+    private List<String> extractBonusIds(SkillVO skillVO) {
```

Comment:
Does it possible to do two operations which performed in two private methods in one stream inside the save method?

Reply:
The idea is in multiple usage of both separated methods. I think one or two of them are also used in update operation

Reply:
The updating method uses the same two methods, maybe we can create only one for those operations?

Reply:
Sure. I just remembered that I did it in the same way is in todo, but here we can create one method instead of two